dispatcher.go

```diff
@@ -351,3 +352,55 @@ func (d *Dispatcher) Stop() error {
 	}
 	return nil
 }
+
+type dispatcherStatus struct {
+	Name       string
+	ID         string
+	Procedures []transport.Procedure
+	Inbounds   []introspection.InboundStatus
+	Outbounds  []introspection.OutboundStatus
+}
+
+func (d *Dispatcher) introspect() dispatcherStatus {
+	var inbounds []introspection.InboundStatus
+	for _, i := range d.inbounds {
+		var status introspection.InboundStatus
+		if i, ok := i.(introspection.IntrospectableInbound); ok {
+			status = i.Introspect()
+		} else {
+			status = introspection.InboundStatus{
+				Transport: "Introspection not supported",
+			}
+		}
+		inbounds = append(inbounds, status)
+	}
+	var outbounds []introspection.OutboundStatus
+	for destService, o := range d.outbounds {
+		var status introspection.OutboundStatus
+		if o.Unary != nil {
+			if o, ok := o.Unary.(introspection.IntrospectableOutbound); ok {
+				status = o.Introspect()
+			} else {
+				status.Transport = "Introspection not supported"
+			}
+			status.Flavor = "unary"
```

Comment:
We've decided that this is called `RPC Type` - I'm fine with switching the terminology, but we'll need to do it everywhere and before 1.0. For now, let's be consistent and use `RPC Type` everywhere.